Remove unused methods in User model

chrisarcand opened the pull request:
Removes three methods which aren't being used anywhere in the app. See commit messages for SHAs where they were introduced.

chrisarcand commented:
@miq-bot add_label core, technical debt

chrisarcand commented:
@Fryguy Hah, one of the methods removed here is one that was edited in #6556 ¯\\\_(ツ)_/¯ Could have been even better the first time!

Fryguy commented on app/models/user.rb:
```diff
@@ -232,10 +232,6 @@ def groups_include?(group)
   end
 
-  def admin?
-    userid == "admin"
```
Are there cases in the UI where the code literally does `== "admin"`.  If so, perhaps it's better to go through this accessor than leave those things in the UI.  The reason I suggest that is that there are known requests around us having the "admin" not be an EVM user, but instead a user that can be pulled in via LDAP.

chrisarcand replied:
There are six places that do that. Sure, I can leave this one then (will change the UI separately).

Fryguy commented on app/models/user.rb:
```diff
@@ -256,14 +252,6 @@ def destroy_widget_sets_for_group(group_id)
   end
 
-  def destroy_orphaned_dashboards
-    (group_ids_of_subscribed_widget_sets - miq_group_ids).each { |group_id| destroy_widget_sets_for_group(group_id) }
```
Are `destroy_widget_sets_for_group` and `group_ids_of_subscribed_widget_sets` (and other methods added for 30cee4e7) needed anymore?

@lfu Can you review this one?

chrisarcand replied:
Good catch @Fryguy, those appear to only be called in spec

chrisarcand commented:
I've rebased and removed more of the methods discussed here.

@lfu Please review the full diff and let me know if any of these methods should not be removed. None of them are called outside of spec. They are being removed because Users and MiqGroups are being split from each other and the less methods I have to 'correct' the better.

Note that the remaining methods around this area are `#destroy_subscribed_widget_sets` and its helper method `#subscribed_widget_sets`. I argue these should be combined and made private as they are only used internally, but I'll leave that for a different PR.

Thanks!

lfu commented:
@chrisarcand @Fryguy 
Looks like these methods for group owned widgets are never used by UI. :+1:

chrisarcand commented:
:tada: Thanks
